### ISSUE TITLE
event_enable_debug_logging missing?

### ISSUE THREAD

**Issue by JoakimSoderberg** [libevent/libevent#12]
I tried enable debugging using event_enable_debug_logging(EVENT_DBG_ALL), but I get a linking error when doing so. I then searched the Libevent code files and couldn't find any implementation of the actual function, it is only present in event.h. Am I missing something vital?

My code:

```
event_enable_debug_mode();
event_enable_debug_logging(EVENT_DBG_ALL);
event_set_log_callback(libeventlog);
```

Where libeventlog does:
    void libeventlog(int severity, const char *msg)
    {
        fprintf(stderr, "\tLIBEVENT: %s\n", msg);
    }

**Comment by zanekarl17** [comment]
I have double and triple checked that I have the latest version of libevent and I keep getting 
"event_enable_debug_logging not declared in this scope" and "EVENT_DBG_ALL not declared in this scope" errors. I have checked the source code in the event2/event.h header that the libevent documentation says this function and macro are contained in and they are not there. Have they been implemented? Are the deprecated and no one documented it? If so what should I use instead of them?

**Comment by azat** [comment]
They are not deprecated, but you can have too damn old version I guess, could you provide next info:
- environment (os/distribution)
- version of libevent

**Comment by azat** [comment]
It was introduced in e30a82f11829e3dc2cf1add253c768fa144db494

**Comment by zanekarl17** [comment]
I appreciate the reply azat, I'm running an Ubuntu 16.04 VM and I have version 2.0.21 of libevent. Any other ideas as to what might be wrong? I've tried updating the libevent-dev package, but that doesn't seem to help at all. Thanks in advance.

**Comment by azat** [comment]
So this fails, yes?

```c
#include <event2/event.h>

int main()
{
    event_enable_debug_logging(EVENT_DBG_ALL);
    return 0;
}
```

```shell
gcc -levent_core test.c || gcc -levent test.c
```

?

**Comment by zanekarl17** [comment]
Yes, this fails.

**Comment by azat** [comment]
Yeah, It was introduced around 2.0.10, but it never merged to 2.0, IOW it is *only in 2.1*, and I'm suggesting you to switch to 2.1, since 2.0 is more-or-less EOL.
